cmd/install/assets/hypershift_operator.go

```diff
@@ -368,11 +369,15 @@ func (o HyperShiftOperatorDeployment) Build() *appsv1.Deployment {
 			envVars = append(envVars,
 				corev1.EnvVar{
 					Name:  "AWS_SHARED_CREDENTIALS_FILE",
-					Value: "/etc/provider/credentials",
+					Value: "/etc/provider/" + o.AWSPrivateSecretKey,
 				},
 				corev1.EnvVar{
 					Name:  "AWS_REGION",
 					Value: o.AWSPrivateRegion,
+				},
+				corev1.EnvVar{
+					Name:  "AWS_SDK_LOAD_CONFIG",
```

Comment:
why this new variable?

Reply:
@jhernand can you elaborate?

Reply:
Apparently the file pointed by `AWS_SHARED_CREDENTIALS_FILE` isn't loaded unless this variable is 1. For example, in my environment if I create a `test.go` file like this:

```go
ackage main

import (
        "fmt"

        "github.com/aws/aws-sdk-go/aws"
        "github.com/aws/aws-sdk-go/aws/session"
        "github.com/aws/aws-sdk-go/service/sts"
)

func main() {
        cfg := aws.NewConfig()
        session, err := session.NewSession(cfg)
        if err != nil {
                panic(err)
        }
        service := sts.New(session)
        response, err := service.GetCallerIdentity(&sts.GetCallerIdentityInput{})
        if err != nil {
                panic(err)
        }
        fmt.Printf("%s\n", *response.Arn)
}
```

And then run it with `AWS_SHARED_CREDENTIALS_FILE` only:

```
AWS_SHARED_CREDENTIALS_FILE=/tmp/credentials go run test.go
```

I get this error:

```
$ AWS_SHARED_CREDENTIALS_FILE=/files/projects/uhc-clusters-service/private-link-creds go run test.go
panic: NoCredentialProviders: no valid providers in chain. Deprecated.
	For verbose messaging see aws.Config.CredentialsChainVerboseErrors

goroutine 1 [running]:
main.main()
	.../test.go:20 +0x12d
```

It works with if the `AWS_SDK_LOAD_CONFIG` variable is added:

```
$ AWS_SDK_LOAD_CONFIG=1 AWS_SHARED_CREDENTIALS_FILE=/files/projects/uhc-clusters-service/private-link-creds go run test.go
arn:aws:sts::...:assumed-role/network-mgmt-.../...
```